src/librustc_interface/passes.rs

```diff
@@ -1040,11 +1040,18 @@ pub fn start_codegen<'tcx>(
 
     // Output Yorick debug sections into binary targets.
     if tcx.sess.crate_types.borrow().contains(&config::CrateType::Executable) {
+        // Are we dumping the textual version of the TIR at the same time?
+        let tir_dump_fn = if tcx.sess.opts.output_types.contains_key(&OutputType::YkTir) {
```

Comment:
`let tir_dump_fn = tcsx.sess...contains_key(...).map(|_| Some(outputs.path(OutputType::YkTir)));` or similar should work.